compiler/rustc_session/src/options.rs

```diff
@@ -1978,6 +1978,10 @@ options! {
         "enable queries of the dependency graph for regression testing (default: no)"),
     randomize_layout: bool = (false, parse_bool, [TRACKED],
         "randomize the layout of types (default: no)"),
+    regparm: Option<u32> = (None, parse_opt_number, [TRACKED],
```

Comment:
This should also include an entry in the [unstable-book](https://github.com/rust-lang/rust/tree/4203c686136428ab10e2765a00886b7c2909a477/src/doc/unstable-book), you can model it after [function-return.md](https://github.com/rust-lang/rust/blob/4203c686136428ab10e2765a00886b7c2909a477/src/doc/unstable-book/src/compiler-flags/function-return.md) or [fixed-x18.md](https://github.com/rust-lang/rust/blob/master/src/doc/unstable-book/src/compiler-flags/fixed-x18.md).
It should link to a tracking issue and describe the intended effects and caveats.